functions/util.cpp

```diff
@@ -5,16 +5,18 @@
 #include <string>
 #include <any>
 
-std::string any_to_str(std::any val) {
-    if (val.type() == typeid(int)) {
-        return std::to_string(std::any_cast<int>(val));
-    } else if (val.type() == typeid(std::string)) {
-        return std::any_cast<std::string>(val);
-    } else if (val.type() == typeid(float)) {
-        return std::to_string(std::any_cast<float>(val));
-    } else if (val.type() == typeid(const char*)) {
-        return std::any_cast<const char*>(val);
-    } else {
-        return "<value>";
+namespace tinygraph {
+    std::string any_to_str(const std::any* val) {
+        if (val->type() == typeid(int)) {
```

Comment:
Use of `typeid` without including `<typeinfo>` makes the program ill-formed.

I assume it was probably transitively included by `<any>` in this case, but standard headers (except for, I think, `<initializer_list>` and `<ios>` etc.) are not required to include any other headers, so one should generally try to explicitly include everything needed.

Reply:
Done